test/test_main.py

```diff
@@ -344,6 +348,32 @@ def test_setup_action_links(self):
         action_hook = self.JUJU_CHARM_DIR / 'actions' / 'test'
         self.assertTrue(action_hook.exists())
 
+    def test_collect_metrics_concurrent(self):
+        indicator_file = self.JUJU_CHARM_DIR / 'indicator'
```

Comment:
That's looks like a very complex test to verify something that is completely unrelated to the way we handle that hook. That's nothing about concurrency that is special for that hook, as I understand it. So what are we testing?  That the framework is opened in exclusive mode?  That should be done in that other PR, and we don't need threads/etc to check that, per today's call. Just open it twice, wait for the error.

Reply:
This was more of a simulation of the existing scenario with Juju to find out where the issues will arise. I created https://github.com/canonical/operator/pull/141 as a result so, as agreed, I can simply remove all the concurrent parts from this one and convert it from the draft.